### DISCUSSION TITLE
PlotPDF function color range for scale bar

### DISCUSSION THREAD

**Discussion by xajezade** [mtex-toolbox/mtex#1794] [Ask Anything]
Hello there,
I am wondering how does color scaling work for the contour plots? I am trying to plot pole figures with different settings for color range and just setting a small value for the min value for the color range makes a lot difference in terms of visuals. I've attached three plots where one is automatic scaling, one is zero for the min of the color range, and one where I've used 0.1 for the min for the color range. Any suggestions on how to select a value for min and max?
automatic scaling:
![plot-automaticScaling](https://user-images.githubusercontent.com/30939214/232633082-12ee6947-3d7d-45a6-a03d-c7e760ff5cff.jpg)
zero as the min value for the color range:
![plot-withZeroAsmin](https://user-images.githubusercontent.com/30939214/232633120-523b6737-6a5b-4e65-bd62-1b71bef59e1f.jpg)
0.1 as the min value for the color range:
![plot-withpointOneasMin](https://user-images.githubusercontent.com/30939214/232633172-748c00d1-33d8-4c84-a471-2d5c34b89884.jpg)

Ali

**Comment by lfgmorales** [comment]
Hi

You can try

clim([cmin cmax])

After your plot

Cheers

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 18 Apr 2023, at 01:38, Ali Khajezade ***@***.***> wrote:

﻿

Hello there,
I am wondering how does color scaling work for the contour plots? I am trying to plot pole figures with different settings for color range and just setting a small value for the min value for the color range makes a lot difference in terms of visuals. I've attached three plots where one is automatic scaling, one is zero for the min of the color range, and one where I've used 0.1 for the min for the color range. Any suggestions on how to select a value for min and max?
automatic scaling:
[plot-automaticScaling]<https://user-images.githubusercontent.com/30939214/232633082-12ee6947-3d7d-45a6-a03d-c7e760ff5cff.jpg>
zero as the min value for the color range:
[plot-withZeroAsmin]<https://user-images.githubusercontent.com/30939214/232633120-523b6737-6a5b-4e65-bd62-1b71bef59e1f.jpg>
0.1 as the min value for the color range:
[plot-withpointOneasMin]<https://user-images.githubusercontent.com/30939214/232633172-748c00d1-33d8-4c84-a471-2d5c34b89884.jpg>

Ali

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1794>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4XTLMOXEFZTMSMKW3XBXIATANCNFSM6AAAAAAXB3JCXE>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1794", "url": "https://github.com/mtex-toolbox/mtex/discussions/1794", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by xajezade** [reply]
Hi 
I think clim only controls the colors on the same contour plot but not the contour plots themselves. 
My question was why we get some strange patterns when I set zero as min of 'colorrange' argument in the PlotPDF function? What's its interpretation? Is it because in automatic scaling it is set for clim to start with zero but the min in the plots are not actually zero?

**Comment by lfgmorales** [comment]
You may want to read about this here

https://mtex-toolbox.github.io/DensityEstimation.html <https://mtex-toolbox.github.io/DensityEstimation.html>

cheers


Luiz
